docs/developer/payments/lifecycle.mdx

```diff
@@ -173,27 +258,126 @@ it may still remain unpaid if the pending amounts fail to process.
 - Status is updated when Checkout's `totalAmount` is changed.
 - Status is updated when Checkout's `TransactionItem`'s amounts are updated.
 
-#### Charge status calculations for Order
 
-The status is calculated based on all transactions assigned to the order.
-It uses the field `chargedAmount` and does not consider **pending amounts** (which differs from the charge status of the checkout).
-The amount used for the status calculation is reduced by `order.totalGrantedRefund`.
+## Transactions recalculation of amounts
 
-- **NONE**: No funds have been charged.
-- **PARTIAL**: The charged funds do not fully cover the amount of `order.total - order.totalGrantedRefund`.
-- **FULL**: The charged funds fully cover the amount of `order.total - order.totalGrantedRefund`.
-- **OVERCHARGED**: The charged funds exceed the amount of `order.total - order.totalGrantedRefund`.
+### Mental model
 
-#### Available transitions
+Each `TransactionItem` has a ledger of events. This way we can calculate current transaction state based on the events history. You can think about it as a bank account: you start with some balance, then you add or subtract amounts based on the events that happen.
 
-- Status is updated when Order's `totalAmount` is changed.
-- Status is updated when Order's `TransactionItem`'s amounts are updated.
-- Status is updated when `Order.grantedRefunds` are updated.
+History cannot be modified, just like in a bank account. In order to reverse an event you must create a new one.
 
-## Transactions recalculation of amounts
+### Order of events
+
+New events can be reported to Saleor in non-subsequent order. This means that some old event might arrive later than a new one. For example:
+- on 10 AM Saleor receives `CHARGE_SUCCESS` event from payment app that was processed at 9:50 AM by payment provider
+- on 10:20 AM Saleor receives `AUTHORIZATION_SUCCESS` event, that was process at 9:30 AM by payment provider.
+
+This will result in a following list of events:
+- `AUTHORIZATION_SUCCESS` with `date: 9:30 AM`
+- `CHARGE_SUCCESS` with `date: 9:50 AM`
+
+### Reporting new events
+
+#### Automatic events
+
+Events are created automatically by Saleor when calling `transactionInitialize` mutation. It creates either:
+- `AUTHORIZATION_REQUEST` when `action` in mutation input or `channel.transactionFlowStrategy` is `AUTHORIZATION`
+- or `CHARGE_REQUEST` when `action` in mutation input or `channel.transactionFlowStrategy` is `CHARGE`
+
+Events are also created when requesting action in Saleor (refund, charge, cancel) via [`transactionRequestAction`](/api-reference/payments/mutations/transaction-request-action.mdx) mutation.
+
+Such events don't have initially any `pspReference`, it's filled in once Saleor receives a response from payment app.
```

Comment:
This can be problematic, in case app returns `CHARGE_ACTION_REQUIRED`, then `CHARGE_REQUEST` with `pspReference`. Saleor will update the `CHARGE_REQUEST` event with returned `pspReference`, but now it will be _before_ the `CHARGE_ACTION_REQUIRED`, which is incorrect.

We have a ticket to look into this: https://linear.app/saleor/issue/SHOPX-1580/transactionprocess-shouldnt-mess-with-transactionitem-event-history

Should we add a note about this in our docs? 🤔

Reply:
Nevermind, I've adedd an example that explains just that.